server/methods/core/groups.js

```diff
@@ -0,0 +1,207 @@
+   * @param {Object} groupData - info about group to create
+   * @param {String} groupData.name - name of the group to be created
+   * @param {Array} groupData.permissions - permissions to assign to the group being created
+   * @param {String} shopId - id of the shop the group belongs to
+   * @return {Object} - object.status of 200 on success or Error object on failure
+   */
+  "group/createGroup": function (groupData, shopId) {
+    check(groupData, Object);
+    check(groupData.name, String);
+    check(groupData.permissions, [String]);
+    check(shopId, String);
+
+    // must have needed permissions.. .TODO: Review who can create a group permission for a shop
+    if (!Roles.userIsInRole(Meteor.userId(), "admin", shopId)) {
+      throw new Meteor.Error(403, "Access Denied");
+    }
+
+    const shop = Shops.update({ _id: shopId }, { $addToSet: { groups: groupData } });
+
+    if (!shop) {
+      throw new Meteor.Error(400, "Bad request data");
+    }
+
+    return { shop, status: 200 };
+  },
+
+  /**
+   * group/updateGroup
+   * @summary updates a permission group for a shop
+   * updates either the name of the permission group or it list of permissions
+   * It also goes into affected user data to modify both the groupName (using Accounts schema)
+   * and group permissions (using "accounts/removeUserPermissions")
+   * @param {Object} group - current data of the group to be updated
+   * @param {String} group.name - name of the group
+   * @param {Array} group.permissions - permissions of the grop
+   * @param {Object} newGroupData - updated group info (similar to current group data)
+   * @param {String} shopId - id of the shop the group belongs to
+   * @return {Object} - object.status of 200 on success or Error object on failure
+   */
+  "group/updateGroup": function (group, newGroupData, shopId) {
+    check(group, Object);
+    check(group.name, String);
+    check(group.permissions, [String]);
+
+    check(newGroupData, Object);
+    check(newGroupData.name, String);
+    check(newGroupData.permissions, [String]);
+    check(shopId, String);
+
+    // must have needed permissions.. .TODO: Review who can create a group permission for a shop
+    if (!Roles.userIsInRole(Meteor.userId(), "admin", shopId)) {
+      throw new Meteor.Error(403, "Access Denied");
+    }
+
+    const permissionsChanged = !_.isEqual(group.permissions, newGroupData.permissions);
+    const updateQuery = { "groups.$": newGroupData };
+
+    Shops.update({ _id: shopId, groups: group }, { $set: updateQuery });
+
+    const matchQuery = { $elemMatch: { shopId: shopId, names: group.name } };
+    const affectedUsers = Accounts.find({ groups: matchQuery }).fetch();
+    let error;
+
+    if (permissionsChanged) {
+      error = updateUsersPermissions(affectedUsers, group, newGroupData, shopId);
+    }
+
+    // update group names on user account
+    error = updateUsersGroupName(affectedUsers, shopId);
```

Comment:
The comment [here](https://github.com/reactioncommerce/reaction/pull/2448/files#r123016688) will likely eliminate the need to do this